packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js

```diff
@@ -106,22 +109,29 @@ function InteractionLayer ({
     }
   }
 
+  function onPointerUp(event) {
+    if (creating) {
+      activeTool.handlePointerUp && activeTool.handlePointerUp(convertEvent(event), activeMark)
+      if (activeMark.finished) onFinish(event)
```

Comment:
UPDATE: OK there's a very odd example of how the  error - or rather, a group of potential errors - can trigger.
- If you create a mark, and you have a **dragging action** that causes the **pointer up event to trigger _outside_ the created mark,** then you've just created an Insidiously Wonky Mark. (IWM)
- I call it _insidious_ because you won't know the mark has broken something until you do something specific to follow up on it.

Example:
1. with macOS10 + Chrome on **touchscreen emulation,** go to https://local.zooniverse.org:8080/ (I Fancy Cats)
2. Answer the "How many cats are there?" and "What are they doing?" questions however you want.
3. On the Drawing Task, select the Point ("mark the cat tails") tool.
4. ❗ With your cursor on any cat's tail, **hold the mouse button down** to create a mark on the cat's tail, and IMMEDIATELY move the cursor elsewhere (e.g. the cat's head) and then **let go of the mouse button**
5. You should see that the mark (actually an IWM) stays at the cat's tail, not the cat's head. Nothing else seems amiss.
6. Now, attempt to delete the mark (by clicking on the 'X' next to the mark)
  - Expected: the mark is deleted
  - Actual: the IWM is deleted, but the error message `Uncaught TypeError: Cannot read property 'finish' of undefined` appears in the console.
  - Any attempts to add more marks will fail, with the same error message.

The above is true for `touch-drawing-events`.

As an added bonus, I also tested this behaviour on `master` and the IWM behaves differently. On `master`, deleting the mark will cause no errors to pop up. Instead, attempting to add a new mark results in the error: `Uncaught TypeError: Cannot read property 'initialDrag' of undefined, at onPointerMove`

Reply:
Yes, something I've noticed on master is that dragging while creating points is broken:
- Expected: the new point mark should follow the pointer and creation should happen on pointer up.
- Actual: the new point doesn't follow the pointer but I do get a new mark wherever pointer up fires.

Reply:
I haven't tried deleting my new points, but it sounds like the delete handling isn't removing `activeMark` from the stored task.

Reply:
I'm trying this out in Firefox. I'm using the point drawing task on the following workflow. Clicking and dragging to create a mark, then deleting the mark, then creating a new mark. I'm not seeing any errors in the console and I am allowed to continue creating new marks. Should I be testing in Chrome?
https://localhost:8080/?project=908&workflow=3370

Reply:
No errors or problems in Chrome either, other than the existing bug on master: moving a mark doesn't re-render the drawing surface. 🤔

Reply:
The code isn't resetting `creating` after removing the active mark in `inactivateMark`. I've added that in, and the errors in touch emulation mode seem to have vanished.

Reply:
I've changed that to use `useEffect`, watching `activeMark` for changes and reacting if it's deleted. It seems to work, testing in Chrome's touch emulator at least. You have to tap twice to create a new point, which is odd, but there are no errors in the console.

Reply:
Woot woot! I can confirm that, as of Commit da773d7, I am NO LONGER able to replicate the IWM error on https://localhost:8080/?project=908&workflow=3370 _nor_ I Fancy Cats, using Chrome on touch emulation (so presumably, any touch device). 👍 

For kicks, I ran tests on 48ec50b, and the IWM error did continue to occur there, so da773d7's addition of `useEffect(function onDeleteMark() { ... }` was indeed the magic bullet.